src/lightning_lite/cli.py

```diff
@@ -13,38 +13,49 @@
 # limitations under the License.
 import logging
 import os
-from argparse import ArgumentParser, Namespace
-from typing import List, Tuple
+from argparse import Namespace
+from typing import Any, List, Optional
+
+from lightning_utilities.core.imports import RequirementCache
 
 from lightning_lite.accelerators import CPUAccelerator, CUDAAccelerator, MPSAccelerator
 from lightning_lite.utilities.device_parser import _parse_gpu_ids
 from lightning_lite.utilities.imports import _IS_WINDOWS, _TORCH_GREATER_EQUAL_1_13
 
 _log = logging.getLogger(__name__)
 
+_CLICK_AVAILABLE = RequirementCache("click")
+
 _SUPPORTED_ACCELERATORS = ("cpu", "gpu", "cuda", "mps", "tpu")
-_SUPPORTED_STRATEGIES = (None, "ddp", "dp", "deepspeed")
+_SUPPORTED_STRATEGIES = ("ddp", "dp", "deepspeed")
 _SUPPORTED_PRECISION = ("64", "32", "16", "bf16")
 
+if _CLICK_AVAILABLE:
```

Comment:
This will incur an import overhead. What if we put this entire definition inside `if name == main`?